### ISSUE TITLE
Poor Indentation Throws Errors

### ISSUE THREAD

**Issue by jpreese** [jwldnr/VisualLinter#12]
### Installed product versions
- Visual Studio: 2017 Professional
- This extension: 1.0.151

### Description
When javascript is not indented properly, the VisualLinter spits out errors in the output window. E.g.

Specified argument was out of the range of valid values.
Parameter name: Start column (?_v1_3996_'{') greater than end column (3990) for line 128
Specified argument was out of the range of valid values.
Parameter name: Start column (?_v1_5148_'{') greater than end column (5142) for line 167
Specified argument was out of the range of valid values.
Parameter name: Start column (?_v1_5269_'{') greater than end column (5263) for line 170

### Steps to recreate
1. Include javascript with poorly indented braces. My exact case was:

```code
        for (var i = 0; i < request.BuildingKey.length; i++) {
        if (add === "") {
            add = add + request.BuildingKey[i];
        } else {
            add = add + ", " + request.BuildingKey[i];
        }
      }
```

Note the last brace actually comes before the for() in terms of spacing

2. Run linter

### Current behavior
Linter displays errors in output window.

![example](https://user-images.githubusercontent.com/6980197/30179961-575b4928-93dc-11e7-9cab-a6d641c41fe6.png)

### Expected behavior
Should not worry about indentation of braces. If anything show linting violations about indentation.

**Comment by jwldnr** [comment]
Thank you for your report @jpreese ,

Unfortunately I am unable to reproduce this on my end.
Is there any chance you could provide me with the full affected `.js file` (where these errors are presented) but also your `ESLint config file`?

**Comment by jpreese** [comment]
I did a little debugging when it happened and it was adding a -1 to EndColumn which would then make StartColumn larger. All of the cases that I saw were do to wonky brace placement.

Though trying to reproduce it now, I'm unable to do so.

**Comment by jwldnr** [comment]
that is because eslint is 1 index based whereas visual studio is 0 based.

I did some searching and found that ESLint sometimes from the result return a larger value start column than end column. this could be to a number of reasons for example if the file has mixed tabs and spaces.

I could try and fix so that this never happens but would need something to test against.
I will keep this issue open so please post back here with the affected `js file` (a gist would be fine), your `.eslintrc` as well as your `package.json` if you encounter this issue again I'll have a look.

**Comment by jwldnr** [comment]
I have tried addressing this issue by making changes to the way this extension generate ranges from ESLint messages.

A new version `1.0.153` will soon be available on the Marketplace. 

Please post back here if updating the extension does not help. I will keep this issue open a while longer.

**Comment by jwldnr** [comment]
Alright, think I've located the bug. The problem was that I ignored the endline and always used startline when eslint reported full range.

Fixed in commit c58ffdba7c23e22cb2416b910a1be0b1cda29c35

A new version `1.0.154` will soon be available on the Marketplace.
